### PR TITLE
chore: add a Taskfile

### PR DESCRIPTION
**Pull request by jgiovaresco**
## Description

[task](https://taskfile.dev/) is a task runner / build tool. It's an alternative to make.

This commit initializes the descriptor with 2 simples tasks:
- build-quick: to build the project without test&validation
- docker: to build Docker images locally

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-kmfjlmlvnv.chromatic.com)

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/add-task-file/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

**Review by ytvnr — APPROVED** [review]
maybe you can add a note on readme on how to use it and if we need to install something ?

File: Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    cmds:
+      - mvn clean install -DskipTests -Dskip.validation -P gio-dev -T 1C
+
+  docker:
+    desc: "Build all 🐳 images"
+    cmds:
+      - docker build -t graviteeio.azurecr.io/apim-gateway:$DOCKER_TAG -f gravitee-apim-gateway/docker/Dockerfile ./gravitee-apim-gateway/gravitee-apim-gateway-standalone/gravitee-apim-gateway-standalone-distribution/target
+      - docker build -t graviteeio.azurecr.io/apim-management-api:$DOCKER_TAG -f gravitee-apim-rest-api/docker/Dockerfile ./gravitee-apim-rest-api/gravitee-apim-rest-api-standalone/gravitee-apim-rest-api-standalone-distribution/target
+      - docker build -t graviteeio.azurecr.io/apim-management-ui:$DOCKER_TAG -f gravitee-apim-console-webui/docker/Dockerfile ./gravitee-apim-console-webui
+      - docker build -t graviteeio.azurecr.io/apim-portal-ui:$DOCKER_TAG -f gravitee-apim-portal-webui/docker/Dockerfile ./gravitee-apim-portal-webui
```
Comment (phiz71) [inline Taskfile.yml:34]: thought (non-blocking): It won't work if build has not been done prior to that task but I'm not sure if we should use `deps` field here.

File: Taskfile.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+version: '3'
+
+env:
+  DOCKER_TAG: local
+
+tasks:
+  build-quick:
+    desc: "Build management-api & gateway quickly without running tests & validations steps"
+    cmds:
+      - mvn clean install -DskipTests -Dskip.validation -P gio-dev -T 1C
```
Comment (phiz71) [inline Taskfile.yml]: question: are you sure of this profile `gio-dev` ? Maybe you mean `distribution-dev` ?

**Review by phiz71 — COMMENTED** [review]
I like the syntax. I find it clearer than a makefile.

**Comment by phiz71** [comment]
> maybe you can add a note on readme on how to use it and if we need to install something ?

We also have this page that we would update.
https://docs.gravitee.io/apim/3.x/apim_devguide_bootstrap.html

File: Taskfile.yml (hunk 2 above)
Comment (jgiovaresco) [reply Taskfile.yml]: That's a profile from `settings.xml` I will use `distribution-dev` instead

File: Taskfile.yml (hunk 1 above)
Comment (jgiovaresco) [reply Taskfile.yml:34]: Maybe we can do that later. Currently, there is no tasks to build the UIs. They have to be built "manually".

Comment (phiz71) [reply Taskfile.yml:34]: Yes, sure. It was just a thought. ;)

File: Taskfile.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+version: '3'
+
+env:
+  DOCKER_TAG: local
```
Comment (ytvnr) [inline Taskfile.yml]:
```suggestion
  DOCKER_TAG: '{{.DOCKER_TAG | default "local"}}'
```

That will allow to run your command like that: `task docker DOCKER_TAG=my-tag`

**Pull request merged**